### PR TITLE
Connect ruleset publish to bwts

### PR DESCRIPTION
**Pull request by niclim**
## 🍗 Description
_What does this PR do? Anything folks should know?_

Connected ruleset publish to bwts and updated the tests

TODO
- [x] Get tests to work

## 📚 References
_Links to relevant docs (Notion, Twist, GH issues, etc.), if applicable._

## 👹 QA
_How can other humans verify that this PR is correct?_

### CODE REVIEW

File: projects/optic/src/__tests__/integration/integration.ts [hunk 1]
```diff
@@ -125,3 +127,40 @@ export async function fileExists(path: string): Promise<boolean> {
 export function normalizeWorkspace(workspace: string, text: string): string {
   return text.replace(new RegExp(workspace, 'g'), '$$workspace$$');
 }
+
+let server: http.Server;
+
+type MockHttpHandler = (req: { url: string; method: string }) => any;
```
Comment (niclim) [inline projects/optic/src/__tests__/integration/integration.ts:133]: Spun up a web service to respond to mock events - takes in a mock handler which can be used to pass in functions, etc

File: projects/optic/src/__tests__/integration/ruleset-publish.test.ts [hunk 2]
```diff
@@ -1,23 +1,36 @@
...
+  runOptic,
+  setupWorkspace,
+  normalizeWorkspace,
+  setupTestServer,
+} from './integration';
 jest.setTimeout(30000);
-// TODO replace this with real mocks when connected to backend
-jest.mock('node-fetch');
+
+setupTestServer(({ url, method }) => {
```
Comment (nalanj) [inline projects/optic/src/__tests__/integration/ruleset-publish.test.ts:9]: Really like this model for testing against a mock server. Pretty smart to just allow it to take in a function that gets the url and method like that.

**Review by nalanj — APPROVED** [review]
Looks great!

**Comment by nalanj** [comment]
Also hooray! I'll be able to just upload rulesets and not have to backchannel inject them into the platform.

**Pull request merged**